spotify.go

```diff
@@ -124,6 +175,74 @@ func decodeError(r io.Reader) error {
 // authenticate, you can use `DefaultClient`.
 type Client struct {
 	http *http.Client
+	retryDuration time.Duration
+}
+
+func (c *Client) ExecuteOpt(req *http.Request, needsStatus int, result interface{}) (err error) {
+	for {
+		resp, err := c.http.Do(req)
+		if err != nil {
+			return err
+		}
+
+		defer resp.Body.Close()
+
+		if resp.StatusCode != http.StatusOK || needsStatus != 0 && resp.StatusCode != needsStatus {
```

Comment:
See above comment. `needsStatus` is provided as 204/`http.StatusCreated` in the couple of cases it is expected.